files/default/plugins/printers.rb

```diff
@@ -0,0 +1,44 @@
+provides 'ohai_gecos'
+
+if ohai_gecos.nil?
+  ohai_gecos Mash.new
+end
+
+begin
+  entries = []
+  printers = []
+
+  entries = Mixlib::ShellOut.new("lpstat -a | awk {'print $1'}")
```

Comment:
Otherwise, collected names could be checked with the exit status of lpoptions/lpq/lpstat for valid printer names. In case of error, the loop should jump to the next (supposed) printer name.